### ISSUE TITLE
use specific version of VSCode

### ISSUE THREAD

**Issue by mikestaub** [codercom/code-server#601]
Is it possible to use a customized version of VSCode? Either to use a specific version or a private fork?

**Comment by kylecarbs** [comment]
Technically, yes. You could fork this repo and create a custom patch.

**Comment by mikestaub** [comment]
Fixed, thanks!

**Comment by mikestaub** [comment]
I would prefer not to fork the repo, but rather be able to create my own dockerfile that inherits from this one. If you explain to me how this can be done, I would be happy to submit a PR.

**Comment by morningspace** [comment]
I have the same question... it looks my current vscode version being used by code-server is 1.33.1... The way that I check the version of vscode is typing `printenv` from the command line within the code-server container, it gives me:
```
TERM_PROGRAM=vscode
TERM_PROGRAM_VERSION=1.33.1
```

However, the latest vscode version at the time of my writing this comment is `1.35.1`. It'd be great that this project can be sync w/ vscode in some way... Haven't had chance dive into the source code, but just did a quick check, it looks there's env var used by build script to define the dependent vscode version, at [this place](https://github.com/cdr/code-server/blob/master/build/tasks.ts#L17). Is that the only place to change if I want to use a specific version of vscode? I feel that would be helpful if that can be documented.

@kylecarbs @mikestaub

**Comment by deansheather** [comment]
It's probably not a good idea to change the version of vscode you use without ensuring the patch is up to date with the changes, since a lot of things can change between vscode versions.

**Comment by morningspace** [comment]
Yeah, that makes sense... just for people who are early adopters :-) If something broken because of incompatible changes introduced that are depended by code-server, then the one who switched the vscode version will have to pay for that :-) On the other hand, it may be also helpful for code-server to apply newer version of vscode to get some sort of early feedback...

**Comment by code-asher** [comment]
Once https://github.com/cdr/code-server/pull/795 is finished it should be possible to use any VS Code version with code-server (it will negate the need for a patch).

**Comment by morningspace** [comment]
That sounds great... looking forward to this feature!

**Comment by jagraj** [comment]
@code-asher 

> Once ~#795~ #857 is finished it should be possible to use any VS Code version with code-server (it will mostly negate the need for a patch).

When this feature is going to be available for the users.? Thank you.
